galaxy_ng/app/migrations/0028_move_perms_to_roles.py

```diff
@@ -0,0 +1,286 @@
+from django.db import migrations
+
+
+MODEL_PERMISSION_TRANSLATOR = [
+    ((
+        ("galaxy", "change_namespace"),
+        ("galaxy", "delete_namespace"),
+        ("galaxy", "view_namespace"),
+        ("galaxy", "upload_to_namespace"),
+        ("ansible", "delete_collection"),
+    ), "galaxy.collection_admin"),
+    ((
+        ("container", "delete_containerrepository"),
+        ("container", "namespace_change_containerdistribution"),
+        ("container", "namespace_modify_content_containerpushrepository"),
+        ("container", "namespace_push_containerdistribution"),
+        ("container", "add_containernamespace"),
+        ("container", "change_containernamespace"),
+    ), "galaxy.execution_environment_admin"),
+    ((
+        ("ansible", "modify_ansible_repo_content"),
+    ), "galaxy.content_admin"),
+    ((
+        ("galaxy", "add_namespace"),
+        ("galaxy", "change_namespace"),
+        ("galaxy", "delete_namespace"),
+        ("galaxy", "view_namespace"),
+        ("galaxy", "upload_to_namespace"),
+        ("ansible", "delete_collection"),
+    ), "galaxy.namespace_owner"),
+    ((
+        ("galaxy", "upload_to_namespace"),
+        ("ansible", "delete_collection"),
+    ), "galaxy.publisher"),
+    ((
+        ("galaxy", "view_group"),
+        ("galaxy", "delete_group"),
+        ("galaxy", "add_group"),
+        ("galaxy", "change_group"),
+    ), "galaxy.group_admin"),
+    ((
+        ("galaxy", "view_user"),
+        ("galaxy", "delete_user"),
+        ("galaxy", "add_user"),
+        ("galaxy", "change_user"),
+    ), "galaxy.user_admin"),
+    ((
+        ("galaxy", "add_synclist"),
+        ("galaxy", "change_synclist"),
+        ("galaxy", "delete_synclist"),
+        ("galaxy", "view_synclist"),
+        ("ansible", "change_collectionremote"),
+    ), "galaxy.synclist_owner"),
+]
+
+
+def batch_process_group_roles(apps, group_roles):
+    GroupRole = apps.get_model("core", "GroupRole")
+    if len(group_roles) > 1000:
+        # print('group_roles > 1000')
+        GroupRole.objects.bulk_create(group_roles)
+        group_roles.clear()
```

Comment:
I'm not sure how expensive the `GroupRole = apps.get_model("core", "GroupRole")` call is. How about the following:

```suggestion
def batch_process(model, objects, flush=False):
    if len(objects) > 1000 or flush:
        # print(f'#{model.__name__} = {len(objects)}')
        model.objects.bulk_create(objects)
        objects.clear()
```

And you would call it with `batch_process(GroupRole, group_roles)` in the loop and once again with `flush=True` at the end. As far as i see, you are currently missing the last unfinished batch.